### PR TITLE
[coq.dev] depends on zarith

### PR DESCRIPTION
**Pull request by liyishuai**
Since coq/coq#11742

Fixes https://github.com/coq/coq/issues/12937

### CODE REVIEW

**Comment by liyishuai** [comment]
Hmm should we have Coq or Bignums to depend on ZArith?

File: core-dev/packages/coq/coq.dev/opam [hunk 1]
```diff
@@ -22,6 +22,7 @@ depends: [
   "ocamlfind" {build}
   "num"
   "conf-findutils" {build}
+  "zarith"    { >= "1.9.1" }
```
Comment (RalfJung) [inline core-dev/packages/coq/coq.dev/opam:25]:
```suggestion
  "zarith" { >= "1.9.1" }
```

Comment (RalfJung) [reply core-dev/packages/coq/coq.dev/opam:25]: Also, why version 1.9.1? The coq docs say ">= 1.8".

Comment (liyishuai) [reply core-dev/packages/coq/coq.dev/opam:25]: Not sure why. I was copying from the opam file in coq/coq.

**Comment by RalfJung** [comment]
@ejgallego  which version of ZArith does Coq need? The docs say 1.8, the opam file says 1.9.1.

**Comment by erikmd** [comment]
Also, shouldn't the `"num"` dependency be removed as well?

**Comment by erikmd** [comment]
On second thought, https://github.com/coq/coq/pull/11742 says `We thus keep the num library linked for now until micromega is ported.`, so forget my previous comment…

**Comment by erikmd** [comment]
@RalfJung 
> which version of ZArith does Coq need? The docs say 1.8, the opam file says 1.9.1.

it seems there is no version 1.8 available in opam-repository:

```
$ opam show zarith
<><> zarith: information on all versions ><><><><><><><><><><><><><><><><><><><>
name                   zarith
[…]
all-versions           1.1  1.2  1.3  1.4  1.4.1  1.5  1.6  1.7  1.7-1  1.9.1
```

**Comment by RalfJung** [comment]
> it seems there is no version 1.8 available in opam-repository:

Interesting.
That would make `>= 1.8` work though... but I guess let's keep this consistent with the file in the Coq repo.

**Pull request merged**